eppo_core/src/events/event_dispatcher.rs

```diff
@@ -34,7 +41,7 @@ pub struct EventDispatcher<T> {
     tx: UnboundedSender<EventDispatcherCommand>,
 }
 
-impl<T: EventQueue + Clone> EventDispatcher<T> {
+impl<T: EventQueue + Clone + Send + 'static> EventDispatcher<T> {
```

Comment:
unclear if this is good practice or not, but since the event queue is used from the delivery async task, it needs to be `'static`